sphinx/writers/latex.py

```diff
@@ -521,6 +521,12 @@ def render(self, template_name: str, variables: dict[str, Any]) -> str:
                                  template_name)
             if path.exists(template):
                 return renderer.render(template, variables)
+            elif template_name.endswith('.jinja'):
+                legacy_template = template[:-len('.jinja')] + '_t'
```

Comment:
Non-obvious detail: the code checks for a `.jinja` suffix on `template_name` -- it should be a short string containing only a filename -- but then we construct the attempted load location from `template` -- a separate string variable that should contain the entire expected path of the (preferred, but not found) template.

Safety/cognitively it might be easier to use the same variable, even if that is slightly less efficient.